src/main/dolphinManager.ts

```diff
@@ -0,0 +1,216 @@
+import { ChildProcessWithoutNullStreams, spawn } from "child_process";
+import { randomBytes } from "crypto";
+import { app } from "electron";
+import electronSettings from "electron-settings";
+import { EventEmitter } from "events";
+import * as fs from "fs-extra";
+import { find, remove } from "lodash";
+import path from "path";
+import { DolphinType, findDolphinExecutable } from "common/dolphin";
+
+electronSettings.configure({
+  fileName: "Settings",
+  prettify: true,
+});
+
+export interface ReplayCommunication {
+  mode: "normal" | "mirror" | "queue"; // default normal
+  replay?: string; // path to the replay if in normal or mirror mode
+  startFrame?: number; // when to start watching the replay
+  endFrame?: number; // when to stop watching the replay
+  commandId?: string; // random string, doesn't really matter
+  outputOverlayFiles?: boolean; // outputs gameStartAt and gameStation to text files (only works in queue mode)
+  isRealTimeMode?: boolean; // default true; keeps dolphin fairly close to real time (about 2-3 frames); only relevant in mirror mode
+  shouldResync?: boolean; // default true; disables the resync functionality
+  rollbackDisplayMethod?: "off" | "normal" | "visible"; // default off; normal shows like a player experienced it, visible shows ALL frames (normal and rollback)
+  queue?: ReplayQueueItem[];
+}
+
+export interface ReplayQueueItem {
+  path: string;
+  startFrame?: number;
+  endFrame?: number;
+  gameStartAt?: string;
+  gameStation?: string;
+}
+
+interface DolphinInstance {
+  dolphinUseType: "playback" | "spectate" | "config" | "netplay";
```

Comment:
Maybe have this as an exported type in `common`?
e.g.
```
export type DolphinType = "playback" | "spectate" | "config" | "netplay";
```
so then you can use it elsewhere to like in line 53.

Or alternatively, even better is probably having this as an enum, like:
```
export enum DolphinType {
  PLAYBACK = "PLAYBACK"
  SPECTATE = "SPECTATE",
  CONFIG = "CONFIG",
  NETPLAY = "NETPLAY",
}
```